docs/reference/commandline/service_create.md

```diff
@@ -177,6 +177,45 @@ $ docker service create \
   redis:3.0.6
 ```
 
+### Container networking
+The containers provisioned in docker swarm mode can be accessed in service discovery either via a Virtual IP (VIP) and routed through the docker swarm ingress overlay network. Or via a DNS round robbin (DNSRR). This configuration is set using the `--endpoint-mode [vip|dnsrr]` setting. Note that dnsrr is incompatible with `--published` ports.
```

Comment:
In general, for the CLI reference, it is better to keep the sections in line with describing and giving examples for various flags, rather than to introduce concepts like **container networking** and **service discovery**.

For example:

### Configure service endpoint mode
Use the `--endpoint-mode`  flag to configure the DNS endpoint mode for containers on an  overlay networking.

`vip` (default): The service name in DNS returns the VIP for the service:

```
$ nslookup service

Address: 10.255.0.7           # this is the VIP on the ingress network

$ nslookup tasks.service
..snip
```

`dnsrr`: The service name in DNS returns the ip addresses for the tasks/containers for the service:

```
$ nslookup service
...result
```